src/huggingface_hub/hf_api.py

```diff
@@ -811,6 +811,15 @@ def create_repo(
         if repo_type not in REPO_TYPES:
             raise ValueError("Invalid repo type")
 
+        if "/" in name:
+            if organization is not None:
+                if organization != name.split("/")[1]:
+                    raise ValueError(
+                        f"Both a `name` with an organization and an `organization` were passed in that do not align ({name}, {organization}). Please ensure these align to the same organization"
+                    )
+                else:
+                    name, organization = name.split("/")
```

Comment:
I would actually do the split first and then check if length is 2 (avoid osan/repo/anotherthing) and raise a more informative value error